chain/indexer/src/streamer/mod.rs

```diff
@@ -77,14 +76,29 @@ async fn build_streamer_message(
         as near_primitives::types::NumShards;
 
     let mut shards_outcomes = fetch_outcomes(&client, block.header.hash).await?;
+    let mut state_changes = fetch_state_changes(
+        &client,
+        block.header.hash,
+        near_primitives::types::EpochId(block.header.epoch_id.clone()),
+    )
+    .await?;
     let mut indexer_shards = (0..num_shards)
-        .map(|shard_id| IndexerShard { shard_id, chunk: None, receipt_execution_outcomes: vec![] })
+        .map(|shard_id| IndexerShard {
+            shard_id,
+            chunk: None,
+            receipt_execution_outcomes: vec![],
+            state_changes: state_changes
+                .remove(&shard_id)
+                .expect("StateChanges for given shard should be present"),
```

Comment:
But some shards (chunks) might be missing in a block, right ?

Reply:
Oh, really, my bad, thanks!